homeassistant/helpers/entity_platform.py

```diff
@@ -126,6 +126,77 @@ async def async_setup_entry(
         """Set up an integration platform from a config entry."""
 
 
+class PlatformData:
+    """Information about a platform, used by entities."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        *,
+        domain: str,
+        platform_name: str,
+    ) -> None:
+        """Initialize the base entity platform."""
+        self.hass = hass
+        self.domain = domain
+        self.platform_name = platform_name
```

Comment:
Should we remove these too from `EntityPlatform` to not have the same data in two places?

Reply:
I think it's a too big breaking change as some non entity integrations are already using it in core and the same could be true for custom integrations. We'd have to look carefully at custom integration usage if we want to do that.